pkg/store/trtl/trtl.go

```diff
@@ -856,8 +857,41 @@ func (s *Store) ListContacts(ctx context.Context) []*models.Contact {
 	return nil
 }
 
-func (s *Store) CreateContact(ctx context.Context, c *models.Contact) (string, error) {
-	return "", nil
+// CreateContact creates a new Contact record in the store, using the contact's
+// email as a unique ID.
+func (s *Store) CreateContact(ctx context.Context, c *models.Contact) (_ string, err error) {
+	if c == nil || c.Email == "" {
+		return "", storeerrors.ErrIncompleteRecord
+	}
+
+	// Marshal the Contact
+	var data []byte
+	if data, err = proto.Marshal(c); err != nil {
+		return "", err
+	}
+
+	ctx, cancel := utils.WithDeadline(ctx)
+	defer cancel()
+
+	// Normalize the email and convert to bytes
+	trimmed := strings.TrimSpace(c.Email)
+	normalized := strings.ToLower(trimmed)
+	key := []byte(normalized)
```

Comment:
One thing that might be helpful would be to add a `Key()` method to the `models.Contact` which returns the key. Then we wouldn't have to have this conversion code in multiple places.